apps/meteor/app/api/server/v1/rooms.ts

```diff
@@ -635,3 +635,37 @@ API.v1.addRoute(
 		},
 	},
 );
+
+API.v1.addRoute(
+	'rooms.isMember',
+	{
+		authRequired: true,
+		validateParams: isRoomsIsMemberProps,
+	},
+	{
+		async get() {
+			const { username, roomId, userId, roomName } = this.queryParams;
```

Comment:
Even if we are doing this already in other endpoints, let's try to keep this simple and use only `userId` and `roomId` it will simplify even the tests we need to do. Also, we should use the `findRoomByIdOrName` function here as it already validates whether the room is archived.

Besides that, since we accepting DM rooms here as well, this could be a problem to search for room names as DMs don't have a name.